release.py

```diff
@@ -0,0 +1,694 @@
+    """Update versions in pyproject.toml."""
+    content = path.read_text()
+    updated = re.sub(rf'version = "{VERSION_REGEX}"', f'version = "{version}"', content)
+    for pkg, pkg_version in deps.items():
+        updated = re.sub(rf'{pkg}=={VERSION_REGEX}', f'{pkg}=={pkg_version}', updated)
+    if content == updated:
+        logger.error(f'No changes made to {path}. Check the versions.')
+        exit(1)
+    path.write_text(updated)
+    logger.info(f'Updated {path} to version {version}')
+
+
+def update_ops_version(ops_version: str, testing_version: str):
+    """Update the ops version in version.py and pyproject.toml."""
+    # version.py
+    ops_src_file_path = VERSION_FILES['ops/src']
+    content = ops_src_file_path.read_text()
+    updated = re.sub(
+        rf"^version: str = '{VERSION_REGEX}'$",
+        f"version: str = '{ops_version}'",
+        content,
+        flags=re.MULTILINE,
+    )
+    ops_src_file_path.write_text(updated)
+    logger.info(f'Updated {ops_src_file_path} to version {ops_version}')
+
+    # pyproject.toml, update both ops-scenario and ops-tracing versions.
+    update_pyproject_versions(
+        VERSION_FILES['ops/pyproject'],
+        testing_version,
+        deps={'ops-scenario': testing_version, 'ops-tracing': ops_version},
+    )
+
+
+def update_testing_version(ops_version: str, testing_version: str):
+    """Update the testing pyproject version."""
+    update_pyproject_versions(VERSION_FILES['testing'], testing_version, deps={'ops': ops_version})
+
+
+def update_tracing_version(ops_version: str):
+    """Update the tracing pyproject version."""
+    update_pyproject_versions(VERSION_FILES['tracing'], ops_version, deps={'ops': ops_version})
+
+
+def update_uv_lock():
+    """Update the uv.lock file with the new versions."""
+    subprocess.run(['uv', 'lock'], check=True)  # noqa: S607
+
+
+def parse_scenario_version():
+    """Parse the current scenario version from pyproject.toml."""
+    file_path = VERSION_FILES['testing']
+    content = file_path.read_text()
+    match = re.search(rf'version = "{VERSION_REGEX}"', content)
+    if not match:
+        raise ValueError(f'Could not find version string in {file_path}')
+    version_str = match.group(1)
+    base_version, dev_suffix = parse_version(version_str)
+    return base_version, dev_suffix
+
+
+def get_new_scenario_version_for_release() -> str:
+    """Get a new version for scenario.
+
+    Default value is generated by removing the ".dev0" suffix from the current scenario version.
+
+    Can be overridden with a user-provided version.
+    """
+    suggested_version, _ = parse_scenario_version()
+    logger.info(f'Suggested new scenario version: {suggested_version}')
+
+    while True:
+        user_input = input(
+            f'Input new scenario version (press enter to use the suggested version '
+            f"{suggested_version or 'required'}), or 'c' to cancel: "
+        ).strip()
+
+        if user_input.lower() == 'c':
```

Comment:
Similar here -- I think we should drop this 'c' handling and just let the user press Ctrl-C.